src/PropertySet.cc

```diff
@@ -722,19 +529,16 @@ void dafBase::PropertySet::copy(std::string const& dest,
                           name + " not in source");
     }
     remove(dest);
-    std::shared_ptr< vector<boost::any> > vp(
-        new vector<boost::any>(*(sj->second)));
-    _set(dest, vp);
-}
-
-/** Appends all value vectors from the \a source to their corresponding
-  * properties.  Sets values if a property does not exist.
-  * @param[in] source PropertySet::Ptr for the source PropertySet.
-  * @throws TypeError Type does not match existing values.
-  * @throws InvalidParameterError Hierarchical name uses non-PropertySet.
-  * @note
-  * May only partially combine the PropertySets if an exception occurs.
-  */
+    if (asScalar) {
+        auto vp = std::make_shared<vector<boost::any>>();
+        vp->push_back(sj->second->back());
+        _set(dest, vp);
+    } else {
+        auto vp = std::make_shared<vector<boost::any>>(*(sj->second));
+        _set(dest, vp);
+    }
+}
```

Comment:
There is a small bit of code duplication here: `_set(det, vp)`. I thought about eliminating it by initializing `vp` to a blank vector outside of the if/else and append to it in the `asScalar` branch and replace it in the `else` branch. I don't have strong feelings either way but felt the current code might be a bit easier to read.

Reply:
I think that's fine.